mesonbuild/compilers/mixins/clike.py

```diff
@@ -1165,19 +1287,21 @@ def thread_flags(self, env):
     def thread_link_flags(self, env: 'Environment') -> T.List[str]:
         return self.linker.thread_flags(env)
 
-    def linker_to_compiler_args(self, args):
-        return args
+    def linker_to_compiler_args(self, args: T.List[str]) -> T.List[str]:
+        return list(args)
```

Comment:
`args.copy()` might be clearer?

Reply:
I need to think about this, it's actually possible that you can get a `CompilerArgs` here... Need to look in the backend code

Reply:
no, this can only be a list, so `.copy()` is right